### PR TITLE
Fix IPv6 handling of cluster IPs in setting.go

### PR DESCRIPTION
**Pull request by thardeck**
## Issue: 

Refers to #43878
 
## Problem
URL generation in`pkg/controllers/dashboard/apiservice/setting.go` does not enclose IPv6 addresses in brackets.
 
## Solution
Check for IPv6 addresses and enclose them in brackets.

### Manual Testing

* Run k3s or rke2 with IPv6 only setup and install the latest Rancher.
* Check out the `fleet-controller` pod and more specifically the `fleet-agentmanagement` container log.
* Make sure that there are no messages like:
```
time="2024-08-22T12:23:47Z" level=error msg="error syncing 'fleet-local/local': handler import-cluster: host must be a URL or a host:port pair: \"https://2001:cafe:43:1::1c8e/k8s/clusters/local\", requeuing"
```

### Automated Testing
* Test types added/modified:
    * Unit

It would be nice to have an IPv6 only integration test for Rancher, but I think this would exceed the scope of this pr/fix., since it most likely requires test infrastructure changes.
Should I create a follow up issue for it?

## QA Testing Considerations
* Besides verifying IPv6 only it might make sense to verify dual stack.
* While searching for other Rancher components having issues with IPv6 I found [pkg/capr/planner/etcdrestore.go](https://github.com/rancher/rancher/blob/58647454340d45abeb9b1bb7f65ba4be344ac7e0/pkg/capr/planner/etcdrestore.go#L320) and fixed it as part of this pr.

  This might need additional testing to verify that the command line parameters with an encapsulated IPv6 are properly escaped and that the changes did not create any regressions.
I was wondering why the issue was not fixed directly  because it was obviously known according to the comments in the code.
* I have also found another IPv6 check in  [pkg/capr/planner/store.go](https://github.com/rancher/rancher/blob/86fdd1de6b2dd7335454fc1c411bb0891fa0b822/pkg/capr/planner/store.go#L566-L571) and changed it to use the same code path.
A unit test for `joinURLFromAddress` does already exist and thus provides extra testing for the new code path.

### CODE REVIEW

File: pkg/settings/setting.go [hunk 1]
```diff
@@ -538,6 +539,14 @@ func IsReleaseServerVersion(serverVersion string) bool {
 	return true
 }
 
+// IsPlainIPV6 will return true if the given address is a plain IPV6 address and not encapsulated or similar.
+func IsPlainIPV6(address string) bool {
+	if net.ParseIP(address) != nil && strings.Count(address, ":") >= 2 {
```
Comment (jakefhyde) [inline pkg/settings/setting.go:544]:
```suggestion
	if ip := net.ParseIP(address); ip != nil && len(ip) == net.IPv6len {
```
May actually be more suitable, but I understand if we don't want to change. FWIW that's how the net package does it: https://cs.opensource.google/go/go/+/refs/tags/go1.23.0:src/net/ip.go;l=229

File: pkg/settings/setting.go [hunk 2]
```diff
@@ -538,6 +539,14 @@ func IsReleaseServerVersion(serverVersion string) bool {
 	return true
 }
 
+// IsPlainIPV6 will return true if the given address is a plain IPV6 address and not encapsulated or similar.
+func IsPlainIPV6(address string) bool {
```
Comment (jakefhyde) [inline pkg/settings/setting.go:543]: I'm not sure if `settings` is the correct package for this, but we don't have a `pkg/util` so maybe `settings` is an okay replacement for that.

**Review by jakefhyde — COMMENTED** [review]
Couple nits

**Review by jakefhyde — APPROVED** [review]

**Comment by snasovich** [comment]
@thardeck , there are conflicts preventing merging this PR (plus we would like +1 approval from Fleet or any other Rancher team). Please rebase, thank you.
cc: @olblak

File: pkg/utils/utils_test.go [hunk 3]
```diff
@@ -29,3 +29,27 @@ func TestFormatPrefix(t *testing.T) {
...
+
+func TestIsPlainIPV6(t *testing.T) {
+	testCases := []struct {
+		ip       string
+		expected bool
+	}{
+		{"::1", true},                         // Loopback IPv6
+		{"2001:cafe:43:1::3ec8", true},        // IPv6 address
+		{"::ffff:7f00:1", true},               // IPv6 address representing IPv4
+		{"192.168.1.1", false},                // IPv4 address
```
Comment (weyfonk) [inline pkg/utils/utils_test.go:41]: Should we add a case for "IPv4 address with port", just to be on the safe side?

**Review by weyfonk — DISMISSED** [review]

**Pull request merged**